### PR TITLE
Node templates: Add debian:bookworm options & remove the ones without OS bit

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Ref: https://github.com/devcontainers/images/issues/611 & https://github.com/devcontainers/images/issues/610

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
-	"image": "mcr.microsoft.com/devcontainers/javascript-node:0-18",
+	"image": "mcr.microsoft.com/devcontainers/javascript-node:1-18-bookworm",
```
Comment (joshspicer) [inline .devcontainer/devcontainer.json:2]: This won't play nice with `docker-in-docker` yet :(

**Review by joshspicer — COMMENTED** [review]
I think we should keep the dev container as is for now (or pin to bullseye)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json:2]: I thought so too, but it strangely worked 😅 

I tested with few things -
- dev container was created fine
- docker works
- devcontainer/cli works

@joshspicer  Should I check something else?

But anyhow, let's switch to `bullseye` to be safe. The current dev container will still point to `bookworm`

Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json]: Updated in [Update devcontainer.json](https://github.com/devcontainers/templates/pull/172/commits/85d40d9aedc2e5661b83ea52978041f81574b421)

**Review by joshspicer — APPROVED** [review]

**Pull request merged**